projects/core/src/cart/occ/occ-cart-payment.adapter.ts

```diff
@@ -0,0 +1,281 @@
+    sopResponseParams['billTo_postalCode'] =
+      fromPaymentProvider[mappingLabels['hybris_billTo_postalcode']];
+
+    sopResponseParams['card_cardType'] = paymentDetails.cardType.code;
+    sopResponseParams['card_accountNumber'] =
+      fromPaymentProvider[mappingLabels['hybris_sop_card_number']];
+    sopResponseParams['card_expirationMonth'] = paymentDetails.expiryMonth;
+    sopResponseParams['card_expirationYear'] = paymentDetails.expiryYear;
+    sopResponseParams['card_nameOnCard'] = paymentDetails.accountHolderName;
+    sopResponseParams['defaultPayment'] = paymentDetails.defaultPayment;
+    sopResponseParams['savePaymentInfo'] = true;
+
+    sopResponseParams['reasonCode'] =
+      fromPaymentProvider[mappingLabels['hybris_sop_reason_code']];
+    sopResponseParams['paySubscriptionCreateReply_subscriptionID'] =
+      fromPaymentProvider[mappingLabels['hybris_sop_subscriptionID']];
+
+    if (mappingLabels['hybris_sop_uses_public_signature'] === 'true') {
+      sopResponseParams[
+        'paySubscriptionCreateReply_subscriptionIDPublicSignature'
+      ] = fromPaymentProvider[mappingLabels['hybris_sop_public_signature']];
+    }
+
+    return sopResponseParams;
+  }
+
+  private getParamsForPaymentProvider(
+    paymentDetails: PaymentDetails,
+    parameters: { key; value }[],
+    mappingLabels: any
+  ) {
+    const params = this.convertToMap(parameters);
+    params[mappingLabels['hybris_account_holder_name']] =
+      paymentDetails.accountHolderName;
+    params[mappingLabels['hybris_card_type']] = paymentDetails.cardType.code;
+    params[mappingLabels['hybris_card_number']] = paymentDetails.cardNumber;
+    if (mappingLabels['hybris_combined_expiry_date'] === 'true') {
+      params[mappingLabels['hybris_card_expiry_date']] =
+        paymentDetails.expiryMonth +
+        mappingLabels['hybris_separator_expiry_date'] +
+        paymentDetails.expiryYear;
+    } else {
+      params[mappingLabels['hybris_card_expiration_month']] =
+        paymentDetails.expiryMonth;
+      params[mappingLabels['hybris_card_expiration_year']] =
+        paymentDetails.expiryYear;
+    }
+    params[mappingLabels['hybris_card_cvn']] = paymentDetails.cvn;
+
+    // billing address
+    params[mappingLabels['hybris_billTo_country']] =
+      paymentDetails.billingAddress.country.isocode;
+    params[mappingLabels['hybris_billTo_firstname']] =
+      paymentDetails.billingAddress.firstName;
+    params[mappingLabels['hybris_billTo_lastname']] =
+      paymentDetails.billingAddress.lastName;
+    params[mappingLabels['hybris_billTo_street1']] =
+      paymentDetails.billingAddress.line1 +
+      ' ' +
+      paymentDetails.billingAddress.line2;
+    params[mappingLabels['hybris_billTo_city']] =
+      paymentDetails.billingAddress.town;
+    params[mappingLabels['hybris_billTo_postalcode']] =
+      paymentDetails.billingAddress.postalCode;
+    return params;
+  }
+
+  private extractPaymentDetailsFromHtml(html: string): any {
```

Comment:
this makes me cry... but i'm sure it's not coming from this PR.

Reply:
Yes, you are right. Fortunately, it is not a part of public API, but only implementation details of OccCartPaymentAdapter